pilot/pkg/model/service.go

```diff
@@ -339,7 +289,7 @@ type ProbeList []*Probe
 // description (which is oblivious to various versions) and a set of labels
 // that describe the service version associated with this instance.
 //
-// Since a ServiceInstance has a single NetworkEndpoint, which has a single port,
+// Since a ServiceInstance has a single IstioEndpoint, which has a single port,
```

Comment:
So the intent was to combine ServiceInstance and the IstioEndpoint into one struct, instead of having multiple levels of hierarchy.. Service has one or more IstioEndpoints (or one or more ServiceInstances). You should be able to get the Endpoint IP, port and other details directly from it. Atlesat, thats what I thought @costinm was hoping to do

Reply:
I know your meaning, but I thought again and again, it is not easy. And I can take two steps. 
First eliminate `NetworkEndpoint`, 
Secondly, try merging `ServiceInstance` and `IstioEndpoint`, I looked into the code, found this step may change some interface return value type and can lead to many changes all over the code.

Reply:
I searched  the whole repo, there are 333 reference of `ServiceInstance`.  So I suggest at least commit a new following up pr, since it may takes long time.